### DISCUSSION TITLE
Set default selected

### DISCUSSION THREAD

**Discussion by tuluong-vl** [rappasoft/laravel-livewire-tables#337] [Q&A] [answered]
Hi there,

Does anyone know the way to set the default selected items for the table?

I can set it by passing the parameter selected to the component

`@livewire('my-table', ['selected' => [1,2,3]]);`

But the check action on the row will not work, it always use the default selected ([1,2,3]) 

Thanks a lot

**Comment by rappasoft** [comment, marked as answer]
You should be able to set default selected right on the table class:

```php
public array $selected = [1,2,3];
```

Where 1,2,3 are the keys of the rows.

**Reply by tuluong-vl** [reply]
Ah sorry it just an example. I mean that i want to pass an array from my view to the component
`@livewire('my-table', ['selected' => $ids]);`

**Reply by tuluong-vl** [reply]
> You should be able to set default selected right on the table class:
> 
> ```
> public array $selected = [1,2,3];
> ```
> 
> Where 1,2,3 are the keys of the rows.

This way same result. Check/uncheck not work for default checked items (always check with any action). In the below example i set default 2 checked row
![Detail](https://media.giphy.com/media/SCs7oJJTyCqIGBDVbN/giphy.gif)

**Reply by tuluong-vl** [reply]
Found the root cause: the **selected**  element value must be **string**
`public array $selected = ["1","2","3"];`
